tc/aten/aten_compiler_new_api-inl.h

```diff
@@ -0,0 +1,145 @@
+/**
```

Comment:
do we really want to have a file called `*_new_api.h`?

Reply:
No we shouldn't but this is the collision-less way of making progress that makes things reviewable.
I wouldn't want to dump the big PR on you ;)
I can rebase and move in the final PR that will switch everything.